### PR TITLE
Add parsing of journalsd configuration to JSON

### PR DESCRIPTION
**Pull request by RamosFe**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/22949|

## Description

Add parsing of `journalsd` configuration to JSON.

### CODE REVIEW

**Review by fdalmaup — CHANGES_REQUESTED** [review]
Related AITs successful execution missing

**Comment by RamosFe** [comment]
## Update
### test_agent_GET_endpoints.tavern.yaml
```bash
(venv) federamos@pop-os:~/Documents/Wazuh/Repositories/wazuh/api/test/integration$ sudo -E ../../../venv/bin/python3 -m pytest test_agent_GET_endpoints.tavern.yaml
========================================================================================================== test session starts ==========================================================================================================
platform linux -- Python 3.9.18, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/federamos/Documents/Wazuh/Repositories/wazuh/api/test/integration
configfile: pytest.ini
plugins: asyncio-0.18.1, html-2.1.1, metadata-3.1.1, tavern-1.23.5, aiohttp-1.0.4, trio-0.8.0
asyncio: mode=auto
collected 95 items                                                                                                                                                                                                                      

test_agent_GET_endpoints.tavern.yaml ...............................................................................................                                                                                              [100%]

=========================================================================================================== warnings summary ============================================================================================================
../../../venv/lib/python3.9/site-packages/_pytest/nodes.py:642
  /home/federamos/Documents/Wazuh/Repositories/wazuh/venv/lib/python3.9/site-packages/_pytest/nodes.py:642: PytestRemovedIn8Warning: The (fspath: py.path.local) argument to YamlFile is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
    return super().from_parent(parent=parent, fspath=fspath, path=path, **kw)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================================================================== 95 passed, 1 warning in 561.59s (0:09:21) ===============================================================================================
```

### test_manager_endpoints.tavern.yaml 
```bash
(venv) federamos@pop-os:~/Documents/Wazuh/Repositories/wazuh/api/test/integration$ sudo -E ../../../venv/bin/python3 -m pytest test_manager_endpoints.tavern.yaml
========================================================================================================== test session starts ==========================================================================================================
platform linux -- Python 3.9.18, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/federamos/Documents/Wazuh/Repositories/wazuh/api/test/integration
configfile: pytest.ini
plugins: asyncio-0.18.1, html-2.1.1, metadata-3.1.1, tavern-1.23.5, aiohttp-1.0.4, trio-0.8.0
asyncio: mode=auto
collected 35 items                                                                                                                                                                                                                      

test_manager_endpoints.tavern.yaml ...................................                                                                                                                                                            [100%]

=========================================================================================================== warnings summary ============================================================================================================
../../../venv/lib/python3.9/site-packages/_pytest/nodes.py:642
  /home/federamos/Documents/Wazuh/Repositories/wazuh/venv/lib/python3.9/site-packages/_pytest/nodes.py:642: PytestRemovedIn8Warning: The (fspath: py.path.local) argument to YamlFile is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
    return super().from_parent(parent=parent, fspath=fspath, path=path, **kw)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================================================================== 35 passed, 1 warning in 739.05s (0:12:19) ===============================================================================================
```

### test_cluster_endpoints.tavern.yaml
```bash
(venv) federamos@pop-os:~/Documents/Wazuh/Repositories/wazuh/api/test/integration$ sudo -E ../../../venv/bin/python3 -m pytest -m cluster test_cluster_endpoints.tavern.yaml 
========================================================================================================== test session starts ==========================================================================================================
platform linux -- Python 3.9.18, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/federamos/Documents/Wazuh/Repositories/wazuh/api/test/integration
configfile: pytest.ini
plugins: asyncio-0.18.1, html-2.1.1, metadata-3.1.1, tavern-1.23.5, aiohttp-1.0.4, trio-0.8.0
asyncio: mode=auto
collected 49 items                                                                                                                                                                                                                      

test_cluster_endpoints.tavern.yaml .................................................                                                                                                                                              [100%]

=========================================================================================================== warnings summary ============================================================================================================
../../../venv/lib/python3.9/site-packages/_pytest/nodes.py:642
  /home/federamos/Documents/Wazuh/Repositories/wazuh/venv/lib/python3.9/site-packages/_pytest/nodes.py:642: PytestRemovedIn8Warning: The (fspath: py.path.local) argument to YamlFile is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
    return super().from_parent(parent=parent, fspath=fspath, path=path, **kw)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================================================================== 49 passed, 1 warning in 918.00s (0:15:17) ===============================================================================================
``

**Review by fdalmaup — APPROVED** [review]
LGTM!

File: framework/wazuh/core/tests/data/configuration/default/journald.conf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<localfile>
+   <location>journald</location>
+   <log_format>journald</log_format>
+   <filter field="MESSAGE_ID">^8d45620c1a4348dbb17410da57c60c66$</filter>
+   <only-future-events>no</only-future-events>
+</localfile>
```
Comment (Selutario) [inline framework/wazuh/core/tests/data/configuration/default/journald.conf:6]:
```suggestion
</localfile>

```

File: framework/wazuh/core/tests/data/configuration/default/journald1.conf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<localfile>
+  <location>journald</location>
+  <log_format>journald</log_format>
+  <filter field="_SYSTEMD_UNIT">^cron.service$</filter>
+  <filter field="PRIORITY" ignore_if_missing="yes">[0-3]</filter>
+</localfile>
```
Comment (Selutario) [inline framework/wazuh/core/tests/data/configuration/default/journald1.conf:6]:
```suggestion
</localfile>

```

**Review by Selutario — CHANGES_REQUESTED** [review]
Good description here: https://github.com/wazuh/wazuh/issues/22949#issuecomment-2073568355

Just a minor request

File: framework/wazuh/core/tests/data/configuration/default/journald.conf (hunk 1 above)
Comment (RamosFe) [reply framework/wazuh/core/tests/data/configuration/default/journald.conf]: Added in ba463c2b3f963a736c4a8f466f6694363fe12bc6.

File: framework/wazuh/core/tests/data/configuration/default/journald1.conf (hunk 2 above)
Comment (RamosFe) [reply framework/wazuh/core/tests/data/configuration/default/journald1.conf]: Added in ba463c2b3f963a736c4a8f466f6694363fe12bc6.

**Review by Selutario — APPROVED** [review]

**Pull request merged**